tests/foreman/api/test_reporttemplates.py

```diff
@@ -162,46 +133,92 @@ def test_positive_report_add_userinput(self):
+        """Lock report template. Check it can be cloned and can't be deleted or edited.
+           Unlock. Check it can be deleted and edited.
 
         :id: a4c577db-144e-4761-a42e-e83887464986
 
         :setup: User with reporting access rights, some report template that is not locked
 
         :steps:
 
-            1. PUT /api/report_templates/:id ... # report_template[locked] = true
+            1. Create template
+            2. Lock template
+            3. Clone template, check cloned data
+            4. Try to delete template
+            5. Try to edit template
+            6. Unlock template
+            7. Edit template
+            8. Delete template
 
         :expectedresults: Report is locked
 
         :CaseImportance: High
-        """
 
-    @tier2
-    @stubbed()
-    def test_positive_unlock_report(self):
-        """Unlock report template
-
-        :id: dae2bff8-340c-4d4c-8349-a2155507a3ab
-
-        :setup: User with reporting and unlock access rights, some report template that is locked
-
-        :steps:
-
-            1. PUT /api/report_templates/:id ... # report_template[locked] = false
-
-        :expectedresults: Report is unlocked
-
-        :CaseImportance: High
+        :BZ: 1680458
         """
+        # Step 1
+        template_name = gen_string('alpha').lower()
+        template_clone_name = gen_string('alpha').lower()
+        template1 = gen_string('alpha')
+        template2 = gen_string('alpha')
+        rt = entities.ReportTemplate(name=template_name, template=template1).create()
+        # Step 2
+        entities.ReportTemplate(id=rt.id, locked=True).update(["locked"])
+        rt = rt.read()
+        self.assertTrue(rt.locked)
+        # Step 3
+        rt.clone(data={'name': template_clone_name})
+        cloned_rt = entities.ReportTemplate().search(
+                  query={'search': u'name="{}"'.format(template_clone_name)})[0].read()
+        self.assertEquals(template_clone_name, cloned_rt.name)
+        self.assertEquals(template1, cloned_rt.template)
+        # Step 4
+        if not is_open('BZ:1680458'):
+            with self.assertRaises(HTTPError):
+                rt.delete()
+            # In BZ1680458, exception is thrown but template is deleted anyway
+            self.assertNotEquals(0, len(entities.ReportTemplate().search(
+                query={'search': u'name="{}"'.format(template_name)})))
+        # Step 5
+        with self.assertRaises(HTTPError):
+            entities.ReportTemplate(id=rt.id, template=template2).update(["template"])
+        rt = rt.read()
+        self.assertEquals(template1, rt.template)
+        # Step 6
+        entities.ReportTemplate(id=rt.id, locked=False).update(["locked"])
+        rt = rt.read()
+        self.assertFalse(rt.locked)
+        # Step 7
+        entities.ReportTemplate(id=rt.id, template=template2).update(["template"])
+        rt = rt.read()
+        self.assertEquals(template2, rt.template)
+        # Step 8
```

Comment:
instead of just putting Step X, could you describe the step? Otherwise, those comments aren't really needed.